discovery/consul/consul.go

```diff
@@ -95,9 +97,25 @@ type SDConfig struct {
 	Scheme       string             `yaml:"scheme,omitempty"`
 	Username     string             `yaml:"username,omitempty"`
 	Password     config_util.Secret `yaml:"password,omitempty"`
+
+	// See https://www.consul.io/docs/internals/consensus.html#consistency-modes,
+	// stale reads are a lot cheaper and are a necessity if you have >5k targets.
+	AllowStale bool `yaml:"allow_stale,omitempty"`
+	// By default use blocking queries (https://www.consul.io/api/index.html#blocking-queries)
+	// but allow users to delay updates if necessary. This can be useful because of "bugs" like
+	// https://github.com/hashicorp/consul/issues/3712 which cause an un-necessary
+	// amount of requests on consul.
+	RefreshInterval model.Duration `yaml:"refresh_interval,omitempty"`
```

Comment:
This is really more of a throttle than the refresh intervals that other SDs have.

Reply:
done "to /delay/throttle/ updates"

Reply:
I meant more the name of the setting. This is not like the refresh_interval setting of other SDs.